test_data/asm/expand_fixed.asm

```diff
@@ -0,0 +1,141 @@
+use std::check::panic;
+
+enum Result<V, E> {
+    Ok(V),
+    Err(E)
+}
+
+/// A term in an array expression
+enum ArrayTerm {
+    Repeat(int[]),
+    Once(int[])
+}
+
+let MORE_THAN_ONE_REPEATED_ERROR = "unsolvable because more than one term is repeated";
+let NON_REPEATED_SIZE_EXCEEDS_DEGREE_ERROR = "non repeated array terms do not fit in degree";
+
+// returns the total size of the repeated array in this array expression
+let compute_length_of_repeated_part: ArrayTerm[], (-> int) -> Result<int, string> = |terms, degree| {
+    let (_, res) = std::array::fold(terms, (false, Result::Ok(0)), |(found_repeated, l), term| {
+        match l {
+            Result::Err(e) => (found_repeated, Result::Err(e)),
+            Result::Ok(len) => match term {
+                // we can have `[]*` as many times as we want
+                ArrayTerm::Repeat([]) => (found_repeated, Result::Ok(len)),
+                ArrayTerm::Repeat(a) => {
+                    if found_repeated {
+                        (true, Result::Err(MORE_THAN_ONE_REPEATED_ERROR))
+                    } else {
+                        (true, Result::Ok(len))
+                    }
+                },
+                ArrayTerm::Once(a) => (found_repeated, Result::Ok(len + std::array::len(a)))
+            }
+        }
+    });
+    match res {
+        Result::Ok(total_size_of_non_repeated) => if total_size_of_non_repeated <= degree() {
+            Result::Ok(degree() - total_size_of_non_repeated)
+        } else {
+            Result::Err(NON_REPEATED_SIZE_EXCEEDS_DEGREE_ERROR)
+        },
+        Result::Err(e) => Result::Err(e)
+    }
+};
+
+enum OffsetSearch {
+    Offset(int),
+    Value(int),
+}
+
+// returns a function representing the array expression
+let expand: ArrayTerm[], (-> int) -> Result<(int -> int), string> = |terms, degree| {
```

Comment:
I found this function quite hard to read... so much nesting, specially in the match.
Maybe some intermediate variables are possible?
Also, we don't support importing enum variants right? 
I think `Ok(None)` would remove some of the noise compared to `Result::Ok(Option::None)`